test-utils/src/address_tree_rollover.rs

```diff
@@ -0,0 +1,356 @@
+use crate::get_hash_set;
+use account_compression::{
+    accounts, initialize_address_merkle_tree::AccountLoader, instruction,
+    utils::constants::ADDRESS_MERKLE_TREE_HEIGHT, AddressMerkleTreeAccount, AddressQueueAccount,
+};
+use anchor_lang::{InstructionData, Key, Lamports, ToAccountMetas};
+use light_concurrent_merkle_tree::ConcurrentMerkleTree;
+use light_hasher::Poseidon;
+use solana_program_test::{
+    BanksClientError, BanksTransactionResultWithMetadata, ProgramTestContext,
+};
+use solana_sdk::{
+    account::{AccountSharedData, WritableAccount},
+    account_info::AccountInfo,
+    instruction::Instruction,
+    pubkey::Pubkey,
+    signature::Keypair,
+    signer::Signer,
+    transaction::Transaction,
+};
+
+pub async fn set_address_merkle_tree_next_index(
+    context: &mut ProgramTestContext,
+    merkle_tree_pubkey: &Pubkey,
+    next_index: u64,
+    lamports: u64,
+) {
+    // is in range 8 -9 in concurrent mt
+    // offset for next index
+    let offset_start = 6 * 8 + 8 + 4 * 32 + 8 * 8;
```

Comment:
memoffset, pls :)

Also, a note for myself mostly - I really need to make the "copy and deserialize account" abstractions less painful and Tokio-compatible. I can totally understand why do you prefer to get data by offsets, but the goal would be to make them robust enough so nobody thinks of doing this. ^